tasks/nsp.js

```diff
@@ -36,8 +41,13 @@ module.exports = function (grunt) {
 
     if (grunt.option('output')) {
       formatter = Nsp.formatters[grunt.option('output')];
-    }
 
+      if (formatters.indexOf(config.output) !== -1) {
+        formatter = Nsp.formatters[grunt.option('output')];
+      } else {
+        grunt.log.write('Invalid formatter specified in options. Must be one of ' + Object.keys(Nsp.formatters).join(',') + '\n');
```

Comment:
You can replace `Object.keys(Nsp.formatters)` here with `formatters` (from line 24).